src/app/shared/doc-viewer/doc-viewer.ts

```diff
@@ -166,4 +170,30 @@ export class DocViewer implements OnDestroy {
     this._clearLiveExamples();
     this._documentFetchSubscription?.unsubscribe();
   }
+
+  _createTooltipsForDeprecated() {
+    // all of the deprecated markers end with `deprecated-marker` 
+    // in their class name
+    const deprecatedElements =
+      this._elementRef.nativeElement.querySelectorAll(`[class$=deprecated-marker]`);
```

Comment:
Is the `$=` needed?

Reply:
its shorter for like:

```js
querySelectorAll('.docs-api-deprecated-marker, .docs-api-class-deprecated-marker, .docs-api-class-interface-marker, .some-more')
```

Reply:
I see. How about adding a small comment illustrating the pattern + using `-deprecated-marker` as suffix test?

Reply:
should be good now, lemme know if it needs rewording cause my english isn't so good.